File: data/transcoder_evaluation_gfg/go/SUM_SERIES_23_45_67_89_UPTO_N_TERMS.go
package main

import (
	"math"
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(n int) float64 {
	var (
		i    int     = 1
		res  float64 = 0.0
		sign bool    = true
	)
	for n > 0 {
		n--
		if sign {
			sign = !sign
			res = res + float64(func() int {
				p := &i
				*p++
				return *p
			}())/float64(func() int {
				p := &i
				*p++
				return *p
			}())
		} else {
			sign = !sign
			res = res - float64(func() int {
				p := &i
				*p++
				return *p
			}())/float64(func() int {
				p := &i
				*p++
				return *p
			}())
		}
	}
	return res
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{49, 4, 60, 90, 96, 29, 86, 47, 77, 87}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if float64(math.Abs(float64(1-(float64(math.Abs(float64(f_gold(param0[i]))))+1e-07)/(float64(math.Abs(float64(f_filled(param0[i]))))+1e-07)))) < 0.001 {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/SUM_SERIES_23_45_67_89_UPTO_N_TERMS.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut n: i32) -> f64 {
    let mut i: i32 = 1 as i32;
    let mut res: f64 = 0.0f64;
    let mut sign: bool = 1 as i32 != 0;
    while n > 0 as i32 {
        n -= 1;
        if sign {
            sign = !sign;
            i += 1;
            i += 1;
            res = res + i as f64 / i as f64
        } else {
            sign = !sign;
            i += 1;
            i += 1;
            res = res - i as f64 / i as f64
        }
    }
    return res;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [49 as i32, 4 as i32, 60 as i32,
         90 as i32, 96 as i32, 29 as i32,
         86 as i32, 47 as i32, 77 as i32,
         87 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if (abs((1 as i32 as f64 -
                     (0.0000001f64 +
                          abs(f_gold(param0[i as usize]) as i32) as
                              f64) /
                         (abs(f_filled(param0[i as usize]) as i32) as
                              f64 + 0.0000001f64)) as i32)
                as f64) < 0.001f64 {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
